apps/crn-server/src/data-providers/contentful/news.data-provider.ts

```diff
@@ -0,0 +1,87 @@
+import { NewsDataObject, NewsFrequency } from '@asap-hub/model';
+
+import {
+  GraphQLClient,
+  parseRichText,
+  FETCH_NEWS,
+  FETCH_NEWS_BY_ID,
+  FetchNewsByIdQuery,
+  FetchNewsByIdQueryVariables,
+  FetchNewsQuery,
+  NewsFilter,
+  NewsOrder,
+  QueryNewsCollectionArgs,
+} from '@asap-hub/contentful';
+
+import { NewsDataProvider, FetchNewsProviderOptions } from '../types';
+
+type NewsItem = NonNullable<
+  NonNullable<FetchNewsQuery['newsCollection']>['items'][number]
+>;
+
+export class NewsContentfulDataProvider implements NewsDataProvider {
+  constructor(private contentfulClient: GraphQLClient) {}
+
+  async fetch(options?: FetchNewsProviderOptions) {
+    const { newsCollection } = await this.contentfulClient.request<
+      FetchNewsQuery,
+      Pick<QueryNewsCollectionArgs, 'limit' | 'skip' | 'order'> & {
+        where?: Partial<NewsFilter>;
+      }
+    >(FETCH_NEWS, {
+      limit: options?.take,
+      skip: options?.skip,
+      order: [NewsOrder.SysFirstPublishedAtDesc],
+      where: {
+        frequency_in: options?.filter?.frequency as string[],
+        title_contains: options?.filter?.title,
```

Comment:
now we have the possible filters appearing in TS as suggestions

<img width="836" alt="types-gql" src="https://user-images.githubusercontent.com/16595804/200308567-5b290518-a621-42a9-bd66-988a0ed5829e.png">